goal_src/engine/gfx/depth-cue.gc

```diff
@@ -5,3 +5,440 @@
+    :front
+    (new 'static 'depth-cue-data :data (new 'static 'vector :x 0.999 :y 0.4 :z 4096.0 :w 1.0))
+    )
+  )
+
+(defun depth-cue-set-stencil ((arg0 dma-buffer) (arg1 int) (arg2 int) (arg3 int) (arg4 dma-gif-packet))
+  (let* ((v1-0 arg0)
+         (t1-0 (the-as object (-> v1-0 base)))
+         )
+    (set! (-> (the-as vector4w t1-0) dword 0) (the-as uint #x10000004))
+    (set! (-> (the-as vector4w t1-0) z) 0)
+    (set! (-> (the-as vector4w t1-0) w) #x50000004)
+    (set! (-> v1-0 base) (&+ (the-as pointer t1-0) 16))
+    )
+  (let* ((v1-1 arg0)
+         (t1-2 (the-as object (-> v1-1 base)))
+         )
+    (set! (-> (the-as vector4w t1-2) dword 0) (make-u128 0 (the-as uint #x3000000000008001)))
+    (set! (-> (the-as vector4w t1-2) dword 1) (the-as uint #xeeeeeeeeeeeeeeee))
+    (set! (-> v1-1 base) (&+ (the-as pointer t1-2) 16))
+    )
+  (let* ((v1-2 arg0)
+         (t1-4 (the-as (pointer uint64) (-> v1-2 base)))
+         )
+    (set! (-> t1-4 0)
+          (the-as uint (logior (shr (shl (+ (* (-> *video-parms* screen-miny) 16) arg3) 48) 16) #x7000))
+          )
+    (set! (-> t1-4 1) (the-as uint 24))
+    (set! (-> t1-4 2) (logior (the-as uint #xffffff00080000) (shr (shl arg1 55) 55)))
+    (set! (-> t1-4 3) (the-as uint 76))
+    (set! (-> t1-4 4) (the-as uint #x50000))
+    (set! (-> t1-4 5) (the-as uint 71))
+    (set! (-> v1-2 base) (&-> t1-4 6))
+    )
+  (let ((a3-7 (the-as (inline-array vector4w) (-> arg0 base)))
+        (v1-3 #x7000)
+        (a1-8 1024)
+        )
+    (set! (-> a3-7 0 quad) (-> *depth-cue-work* stencil-tmpl dma-vif quad))
+    (set! (-> a3-7 1 quad) (-> *depth-cue-work* stencil-tmpl quad 1))
+    (let ((v0-0 (the-as object (-> a3-7 2))))
+      (dotimes (a3-8 2)
+        (set! (-> (the-as (inline-array vector4w) v0-0) 0 quad) (-> arg4 dma-vif quad))
+        (set! v0-0 (-> (the-as (inline-array vector4w) v0-0) 1))
+        (dotimes (t1-10 5)
+          (set-vector! (-> (the-as (inline-array vector4w) v0-0) 0) v1-3 (* (-> *video-parms* screen-miny) 16) arg2 0)
+          (set-vector!
+            (-> (the-as (inline-array vector4w) v0-0) 1)
+            (+ v1-3 a1-8)
+            (* (-> *video-parms* screen-maxy) 16)
+            arg2
+            #x10000
+            )
+          (+! v1-3 a1-8)
+          (set! v0-0 (-> (the-as (inline-array vector4w) v0-0) 2))
+          )
+        )
+      (set! (-> arg0 base) (the-as pointer v0-0))
+      (the-as vector4w v0-0)
+      )
+    )
+  )
+
+(defun depth-cue-draw-depth ((arg0 dma-buffer) (arg1 int) (arg2 float) (arg3 float) (arg4 int) (arg5 int))
+  (set! (-> *depth-cue-work* draw-color w) (the int (* 128.0 arg3)))
+  (let ((v1-1 (the int (* 512.0 arg2)))
+        (a2-1 (the int (* (the float (* (-> *video-parms* screen-sy) 16)) arg2)))
+        (a3-4 0)
+        (t2-0 512)
+        (t3-0 *depth-cue-base-page*)
+        )
+    (dotimes (t4-0 16)
+      (let* ((t5-0 arg0)
+             (t6-0 (the-as object (-> t5-0 base)))
```

Comment:
Same ideas as before. A DMA transfer to the GIF: a `dma-packet`, a `gif-tag` and then the parameters.